nengo_loihi/compat.py

```diff
@@ -111,3 +25,30 @@ def make_process_step(process, shape_in, shape_out, dt, rng, dtype=None):
     nengo_dl = None
     HAS_DL = False
     logger.debug("Error importing NengoDL:\n%s", err)
+
+
+def is_transform_type(transform, types):
+    types = (types,) if isinstance(types, str) else types
+    types = tuple(
+        getattr(nengo.transforms, t) for t in types if hasattr(nengo.transforms, t)
+    )
+    return isinstance(transform, types)
+
+
+def sample_transform(conn, rng=np.random):
+    if is_transform_type(conn.transform, "NoTransform"):
+        return np.array(1.0)
+
+    transform = conn.transform.sample(rng=rng)
+
+    # convert SparseMatrix to scipy.sparse
+    if isinstance(transform, nengo.transforms.SparseMatrix):
+        transform = scipy.sparse.csr_matrix(
+            (transform.data, transform.indices.T), shape=transform.shape
+        )
+    return transform
+
+
+def make_process_step(process, shape_in, shape_out, dt, rng, dtype=None):
+    state = process.make_state(shape_in, shape_out, dt, dtype=dtype)
+    return process.make_step(shape_in, shape_out, dt, rng, state)
```

Comment:
It's kind of awkward having these functions in `compat.py`, since they aren't really related to version compatibility any more. But I couldn't think of a place to put them that wasn't also awkward, so 🤷.

Reply:
I agree. Some of them, like this one, could probably just be gotten rid of entirely (I just didn't spend too much time on it).